File: sequential/C++/native_stitching.cpp
#include <opencv2/core/core.hpp>
#include <opencv2/features2d/features2d.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/calib3d/calib3d.hpp>
#include <opencv2/stitching.hpp>

#include <iostream>

using namespace std;
using namespace cv;

int main()
{
    vector<Mat> imgs;
    int numOfImages = 4;
    const char *images[numOfImages] = {"images/1.jpg", "images/2.jpg", "images/3.jpg", "images/4.jpg"};

    for (int i = 0; i < numOfImages; i++) {
        Mat img = imread(images[i]);
        imgs.push_back(img);
    }

    Mat pano;
    Stitcher::Mode mode = Stitcher::PANORAMA;
    Ptr<Stitcher> stitcher = Stitcher::create(mode);
    Stitcher::Status status = stitcher->stitch(imgs, pano);
    imwrite("images/native_result.jpg", pano);
}

File: sequential/python/native_stitching.py
import cv2
import time

image_paths=['images/1.jpg', 'images/2.jpg', 'images/3.jpg', 'images/4.jpg']

imgs = []
for i in range(len(image_paths)):
    imgs.append(cv2.imread(image_paths[i]))

stitcher = cv2.Stitcher.create()

start = time.time()

(res, output) = stitcher.stitch(imgs)

end = time.time()

print("Computational Time: " + str(end - start))

if res == cv2.STITCHER_OK:
    cv2.imwrite("images/result.jpg", output)